### ISSUE TITLE
Please keep empty lines. Don't remove it without explicitly action.

### ISSUE THREAD

**Issue by code-brewer** [marktext/marktext#3455]
I found two fact:

1)  marktext won't show empty lines of files that created by 'Typora'

2) If insert empty lines by editing, they are deleted once saved.

It is poor readability if no empty and all lines goes without space. 

Please let empty lines be. Give the choice to user to determine if delete empty lines or not!

**Comment by YalandHong** [comment]
#1354

**Comment by symonxdd** [comment]
Yes! I can't believe this repo is so popular and yet lacks a critical feature smh

**Comment by jpggithub** [comment]
Some presentations with reveal.js and markdown files need to have (for example) two or three empty lines to be well displayed (to separate the different slides).
Yes, it will be a normal mode to not modify a file if we don't ask for it.
Currently, if I load a file with several empty lines at one place and want to close it without touching anything, Marktext ask if I want to save it. Surprising!

**Comment by ControlCoreX** [comment]
I am also being bothered now, it will secretly modify my files, and I don't want it to be like this.

**Comment by Jocs** [comment]
Closing as a duplicate of #1354, which tracks MarkText collapsing consecutive blank lines because the document is stored as a block structure that does not record how many blank lines sat between blocks. It has the maintainers' root-cause explanation and the discussion about a preference to control it.

Two things from this thread are worth carrying over to #1354:

- The reveal.js case. Slide separators need two or three consecutive blank lines, so "preserve exactly N" matters — "preserve one" would not be enough.
- Being prompted to save a file you never edited is the same rewrite seen from the other end; that half is tracked in #2189.
